components/bl-leafletMap-component/src/index.js

```diff
@@ -0,0 +1,115 @@
+import { useEffect, useRef } from 'react';
+
+import {
+  initMap,
+  createMarkers,
+  createCircles,
+  createPolygons,
+  changeMapType,
+  setCenter,
+  setZoom,
+  toggleZoomControl,
+  toggleDraggingControl,
+  toggleFullscreen
+} from './helpers/map';
+import { createActions } from './helpers/actions';
+import { useIcon } from './helpers/use-icon';
+import { IconOptions } from './icons';
+import { MapTypeSelect } from './map-type-select';
+import { GeopositionButton } from './geoposition-button';
+import { FullscreenButton } from './fullscreen-button';
+
+export default function LeafletMap({ component, eventHandlers }) {
+  const {
+    markers,
+    circles,
+    polygons,
+    center,
+    zoom,
+    mapType,
+    classList,
+    visibility,
+    zoomControl,
+    draggingControl,
+    mapTypeControl,
+    geopositionControl,
+    display,
+    fullscreen,
+    fullscreenControl
+  } = component;
+
+  const containerRef = useRef(null);
+  const mapRef = useRef(null);
+  const currentLayer = useRef(null);
+  const markerIcon = useIcon(IconOptions.marker);
+
+  useEffect(() => {
+    initMap(component, eventHandlers, mapRef, currentLayer);
+    createActions(component, mapRef.current);
+  }, []);
+
+  useEffect(() => {
+    changeMapType(mapRef, currentLayer, component);
+  }, [mapType]);
+
+  useEffect(() => {
+    createMarkers(markers, markerIcon, mapRef.current, eventHandlers);
+  }, [markers]);
+
+  useEffect(() => {
+    createCircles(circles, mapRef.current, eventHandlers);
+  }, [circles]);
+
+  useEffect(() => {
+    createPolygons(polygons, mapRef.current, eventHandlers);
+  }, [polygons]);
+
+  useEffect(() => {
+    setCenter(mapRef.current, center);
+  }, [center]);
+
+  useEffect(() => {
+    setZoom(mapRef.current, zoom);
+  }, [zoom]);
+
+  useEffect(() => {
+    toggleZoomControl(mapRef.current, zoomControl);
+  }, [zoomControl]);
+
+  useEffect(() => {
+    toggleDraggingControl(mapRef.current, draggingControl);
+  }, [draggingControl]);
+
+  useEffect(() => {
+    toggleFullscreen(containerRef.current, fullscreen, mapRef.current);
+  }, [fullscreen]);
+
+  if (!display || !visibility) {
+    return null;
+  }
+
+  return (
+    <div
+      id="bl-customComponent-leafletMap"
```

Comment:
what's the id for?

Reply:
export function initMap(component, eventHandlers, map, currentLayer) {
  const { zoom, center, mapType, zoomControl } = component;
  const { onClick } = eventHandlers;
  const centerCoords = toCoordinates(center);

  map.current = Leaflet.map('bl-customComponent-leafletMap', {
    zoom,
    center           : centerCoords,
    fullscreenControl: true
  });
  
  This id need for init map , first arg in Leaflet.map() - id

Reply:
I solved this problem by accepting id as a property